pep-0680.rst

```diff
@@ -0,0 +1,501 @@
+we are not aware of any use cases for it.
+
+Controlling the type of mappings returned by ``tomllib.load[s]``
+----------------------------------------------------------------
+
+The ``toml`` library on PyPI supports a ``_dict`` argument, which works
+similarly to the ``object_hook`` argument in ``json.load[s]``. There are several
+uses of ``_dict`` found on https://grep.app, however, almost all of them are
+passing ``_dict=OrderedDict``, which should be unnecessary as of Python 3.7. We
+found two instances of legitimate use: in one case, a custom class was passed
+for friendlier KeyErrors, in another case, the custom class had several
+additional lookup and mutation methods (e.g. to help resolve dotted keys).
+
+Such an argument is not necessary for the core use cases outlined in the
+motivation section. The absence of this can be pretty easily worked around using
+a wrapper class, transformer function, or a third-party library. Finally,
+support could be added later in a backward compatible way.
+
+
+Removing support for ``parse_float`` in ``tomllib.load[s]``
+-----------------------------------------------------------
+
+This option is not strictly necessary, since TOML floats are "IEEE 754 binary64
+values", which is ``float`` on most architectures. Using ``decimal.Decimal``
+thus allows users extra precision not promised by the TOML format. However, in
+the author of ``tomli``'s experience, this is useful in scientific and financial
+applications. TOML-facing users may include non-developers who are not aware of
+the limits of double-precision float.
+
+There are also niche architectures where the Python ``float`` is not a IEEE-754
+binary64. The ``parse_float`` argument allows users to achieve correct TOML
+semantics even on such architectures.
+
+
+Alternative names for module
+----------------------------
+
+Ideally, we would be able to use the ``toml`` module name.
+
+However, the ``toml`` package on PyPI is widely used, so there are backward
+compatibility concerns. Since the standard library takes precedence over third
+party packages, users who have pinned versions of ``toml`` would be broken when
+upgrading Python versions by any API incompatibilities.
+
+To further clarify, the user pins are the specific concern here. Even if we were
+able to get control over the ``toml`` PyPI package and repurpose it as a
+standard library backport, we would still break users who have pinned to
+versions of the current ``toml`` package. This is unfortunate, since pinning
+would likely be a common response to breaking changes introduced by repurposing
+the ``toml`` package as a backport (that is incompatible with today's ``toml``).
+
+There are several API incompatibilities between ``toml`` and the API proposed in
+this PEP, listed in `Appendix A`_.
+
+Finally, the ``toml`` package on PyPI is not actively maintained and `we have
+been unable to contact the author <https://github.com/uiri/toml/issues/361>`,
```

Comment:
underscores are missing to make this a link

Reply:
Addressed this here:  `https://github.com/python/peps/pull/2223`